CONTRIBUTING.md

```diff
@@ -0,0 +1,481 @@
+the master prettier formatting. If your branch/code does NOT meet the prettier
+formatting requirements, it will not be merged and it will delay its review.
+
+If your editor is not configured to auto format on save using prettier, then you
+can run the following command to auto format all files in your local repo/PR:
+
+```shell
+yarn prettier:fix
+```
+
+You can also run the prettier check command to see which files do not follow the
+prettier formatting guidelines.
+
+```shell
+yarn prettier
+```
+
+## Content types
+
+This repo contains multiple different types of developer content records. Each
+grouping of records (called a "content type") serve a different purpose and are
+handled differently when displayed in the UI of `solana.com`.
+
+Below is a table describing each active content type group, including the
+corresponding path within this repo and webpage for viewing on solana.com:
+
+| Content Type | Repo Path                                   | Webpage URL                                               |
+| ------------ | ------------------------------------------- | --------------------------------------------------------- |
+| `docs`       | [`/docs`](/docs/)                           | [View Docs](https://solana.com/docs)                      |
+| `guides`     | [`/content/guides`](/content/guides/)       | [View Guides](https://solana.com/developers/guides)       |
+| `resources`  | [`/content/resources`](/content/resources/) | [View Resources](https://solana.com/developers/resources) |
+| `cookbook`   | [`/content/cookbook`](/content/cookbook)    | [View Cookbook](https://solana.com/developers/cookbook)   |
+
+## Written in markdown
+
+Every piece of content within this repo is normally written in markdown with
+yaml frontmatter. With some support for specific
+[custom components](#components) using React and MDX.
+
+We typically use GitHub flavor markdown (GFM) which you can
+[learn more about here](https://github.github.com/gfm/).
+
+## Frontmatter
+
+The yaml frontmatter is used to provide additional metadata about a given piece
+of content. Each content type has specifically supported frontmatter fields,
+some required and some optional. All available frontmatter fields are enforced
+via [Contentlayer](#why-contentlayer) (see more below).
+
+### Why Contentlayer?
+
+[Contentlayer](https://contentlayer.dev/) offers yaml frontmatter structure
+enforcement and type safety for the frontmatter fields on every content record,
+including generating TypeScript types for each content type's grouping.
+
+Contentlayer does this code generation by enabling us to define a custom data
+schema for the frontmatter. You can view the current Contentlayer schema here:
+[`contentlayer.config.ts`](/contentlayer.config.ts)
+
+If any content records contains unsupported or misspelled fields in the
+frontmatter, Contentlayer will throw an error. Preventing misconfigured content
+from being shipped to production.
+
+### Default frontmatter fields
```

Comment:
maybe a link to one of the pages so readers have a reference for what the frontmatter is supposed to look like, or add a snippet for each field